airbyte-integrations/connectors/destination-mongodb/src/main/java/io/airbyte/integrations/destination/mongodb/MongodbDestination.java

```diff
@@ -114,51 +121,53 @@ public AirbyteMessageConsumer getConsumer(JsonNode config,
 
   /* Helpers */
 
-  private MongoClientURI getMongoClientURI(JsonNode config) {
-    MongoClientURI uri = null;
-    if (config.get(AUTH_TYPE).get(AUTHORIZATION).asText().equals(LOGIN_AND_PASSWORD)) {
-      uri = new MongoClientURI(String.format(MONGO_URI, config.get(AUTH_TYPE).get(USERNAME).asText(),
-          config.get(AUTH_TYPE).get(PASSWORD).asText(), config.get(HOST).asText(),
-          config.get(PORT).asText(), config.get(DATABASE).asText()));
-    }
-    return uri;
+  private MongoDatabase getDatabase(JsonNode config) {
+    return new MongoDatabase(getConnectionString(config), config.get(DATABASE).asText());
   }
 
-  private MongoClient getMongoClient(String host, int port, @Nullable MongoClientURI uri) {
-    try {
-      if (uri == null) {
-        var serverAddress = new ServerAddress(host, port);
-        return new MongoClient(serverAddress);
-      } else {
-        return new MongoClient(uri);
-      }
-    } catch (RuntimeException e) {
-      throw new RuntimeException(e);
+  @VisibleForTesting
+  String getConnectionString(JsonNode config) {
+    var credentials = config.get(AUTH_TYPE).get(AUTHORIZATION).asText().equals(LOGIN_AND_PASSWORD)
+        ? String.format("%s:%s@", config.get(AUTH_TYPE).get(USERNAME).asText(), config.get(AUTH_TYPE).get(PASSWORD).asText())
+        : StringUtils.EMPTY;
+
+    // backward compatibility check
```

Comment:
Nitpick. This comment is very important. Thank you for adding this.

However, it is not clear which part in the `if` block is the old vs the new configs. Can you add more comments to clarify it? Something like:

```suggestion
    // backward compatibility check
    // the old mongo db spec only includes host, port, database, and auth_type
    // the new spec replaces host and port with the instance_type property
```

Reply:
Sorry that I missed this in the first round of review.